go/vt/vtgate/vtgate.go

```diff
@@ -58,47 +58,80 @@ import (
+	lockHeartbeatTime *time.Duration
+	warnShardedOnly   *bool
 
-	foreignKeyMode = flag.String("foreign_key_mode", "allow", "This is to provide how to handle foreign key constraint in create/alter table. Valid values are: allow, disallow")
+	foreignKeyMode *string
 
 	// flags to enable/disable online and direct DDL statements
-	enableOnlineDDL = flag.Bool("enable_online_ddl", true, "Allow users to submit, review and control Online DDL")
-	enableDirectDDL = flag.Bool("enable_direct_ddl", true, "Allow users to submit direct DDL statements")
+	enableOnlineDDL *bool
+	enableDirectDDL *bool
 
-	enableSchemaChangeSignal = flag.Bool("schema_change_signal", true, "Enable the schema tracker; requires queryserver-config-schema-change-signal to be enabled on the underlying vttablets for this to work")
-	schemaChangeUser         = flag.String("schema_change_signal_user", "", "User to be used to send down query to vttablet to retrieve schema changes")
+	enableSchemaChangeSignal *bool
+	schemaChangeUser         *string
 )
 
+func RegisterFlags(fs *pflag.FlagSet) {
+	fs.StringVar(transactionMode, "transaction_mode", "MULTI", "SINGLE: disallow multi-db transactions, MULTI: allow multi-db transactions with best effort commit, TWOPC: allow multi-db transactions with 2pc commit")
+	fs.BoolVar(normalizeQueries, "normalize_queries", true, "Rewrite queries with bind vars. Turn this off if the app itself sends normalized queries with bind vars.")
+	fs.BoolVar(terseErrors, "vtgate-config-terse-errors", false, "prevent bind vars from escaping in returned errors")
+	fs.IntVar(streamBufferSize, "stream_buffer_size", 32*1024, "the number of bytes sent from vtgate for each stream call. It's recommended to keep this value in sync with vttablet's query-server-config-stream-buffer-size.")
+	fs.Int64Var(queryPlanCacheSize, "gate_query_cache_size", cache.DefaultConfig.MaxEntries, "gate server query cache size, maximum number of queries to be cached. vtgate analyzes every incoming query and generate a query plan, these plans are being cached in a cache. This config controls the expected amount of unique entries in the cache.")
+	fs.Int64Var(queryPlanCacheMemory, "gate_query_cache_memory", cache.DefaultConfig.MaxMemoryUsage, "gate server query cache size in bytes, maximum amount of memory to be cached. vtgate analyzes every incoming query and generate a query plan, these plans are being cached in a lru cache. This config controls the capacity of the lru cache.")
+	fs.BoolVar(queryPlanCacheLFU, "gate_query_cache_lfu", cache.DefaultConfig.LFU, "gate server cache algorithm. when set to true, a new cache algorithm based on a TinyLFU admission policy will be used to improve cache behavior and prevent pollution from sparse queries")
+	_ = fs.Bool("disable_local_gateway", false, "deprecated: if specified, this process will not route any queries to local tablets in the local cell")
```

Comment:
if this has been deprecated, can we go ahead and delete it? (i'm assuming this was deprecated in v14)